lib/url.js

```diff
@@ -1,6 +1,14 @@
 'use strict';
 
-const punycode = require('punycode');
+function importPunycode() {
+  try {
+    return process.binding('icu');
+  } catch (e) {
+    return require('punycode');
+  }
+}
+
+const {toASCII} = importPunycode();
```

Comment:
should we have a spacing rule for destructuring before we start using it in full? // @Trott